pkg/app/piped/controller/controller.go

```diff
@@ -369,6 +371,32 @@ func (c *controller) syncPlanners(ctx context.Context) error {
 	}
 
 	for appID, d := range pendingByApp {
+		plannable, err := c.shouldStartPlanningDeployment(ctx, d)
+		if err != nil {
+			c.logger.Error("failed to check deployment plannability",
+				zap.String("deployment", d.Id),
+				zap.String("app", d.ApplicationId),
+				zap.Error(err),
+			)
+			continue
+		}
+
+		if !plannable {
+			if d.IsInChainDeployment() {
+				c.logger.Info("unable to start planning deployment, probably locked by the previous block in its deployment chain",
+					zap.String("deployment_chain", d.DeploymentChainId),
+					zap.String("deployment", d.Id),
+					zap.String("app", d.ApplicationId),
+				)
+			} else {
+				c.logger.Info("unable to start planning deployment, try again next sync interval",
+					zap.String("deployment", d.Id),
+					zap.String("app", d.ApplicationId),
+				)
+			}
+			continue
+		}
+
```

Comment:
Do we have any way to show the reason why a deployment has not been planned on the web?
Should we set that message by `trigger` or `planner`?

Reply:
Currently, we don't have, since the state of the deployment is not changed, so the message for that `keep PENDING` may be redundant to me. If we want to show kind of reason why the deployment keeps pending, we may need a new RPC like UpdateDeploymentStatusReason or such. How do you think?

Reply:
I just wanted to ask to know did you have any plan for that thing yet.
Could you create a new issue to ensure that it could not be flushed out of our memory over time?
Btw, I will approve this PR since that is out of the scope of this PR.

Reply:
I see, in that case

> Should we set that message by trigger or planner?

Probably not the `trigger` since after triggering the deployment to start (PENDING state) the trigger is out of action, and this `check is plannable` logic run in syncPlaner phase but before a planner starts running, so the one who submits that message should be controller itself, via kind of UpdateDeploymentStatusReason RPC (the deployment state is unchanged but update the desc?) that is in my image. Let's me think about that in detail later (will create issue for that)

Reply:
created https://github.com/pipe-cd/pipe/issues/2848 🙏